### PR TITLE
Admin settings

### PR DESCRIPTION
**Pull request by trdtnguyen**
- [x ] The changes don't come from your master branch. ([Source](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)) 
---

### [Trello Card](https://trello.com/b/vi6WuHYk/admin-portal)
Trello card: "Setting Pages backend support"
#### What kind of change does this PR introduce? 
Implement the Admin Setting page in the back-end.
(1) Add the corresponding APIs in `config/routes.rb`.
(2) Add admin controller `app/controllers/admins_controller.rb` to handle the APIs.
(3) Modify the admin model `app/models/admin.rb` to validate input data for admin

*** Updated on the second revision ***
(4) Fix bugs in `app/controller/admin_controllers`
(5) Change the name of function `current_donor` to `current_user` so that the function works for admin, donor, and client.
(6) Add unit tests for the admin's controller.

Front-end could save settings in the Admin page using `admins/:id/update` API.
#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
Notes:
(1) This PR also adds API for creating admin. So back-end devs who are working with admins table should carefully check this to avoid duplicate code.
(2) This PR modifies the `app/controllers/application_controller.rb` that used as a global shared between admin, donor, and client.

#### Images (before/ after screenshots, interaction GIFs, ...)

---

#### TODO
 - Handle profile picture
- Corresponding front-end testing is needed.

### CODE REVIEW

File: app/controllers/admins_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    skip_before_action :authorized, only: [:create]
+
+	def create
+		return render json: { error: 'admin email already in use'}, status: :conflict if Admin.exists?({email: admin_params[:email]})
+		@admin = Admin.create(admin_params)
+		if @admin.valid?
+			@token = encode_token(admin_id: @admin.id)
+			session[:admin_id] = @admin.id
+
+			if @admin.save
```
Comment (sagehen03) [inline app/controllers/admins_controller.rb:11]: I don't think you need to call save here.  `@admin = Admin.create(admin_params)` implies saving to the database if `@admin.valid?` at least that's my understanding.

File: config/routes.rb [hunk 2]
```diff
@@ -25,6 +25,8 @@
   get 'clients/:id/claims_history', to: 'clients#claims_history'
 
   post 'admin_auth', to: 'admin_auth#create'
+  patch 'admins/create', to: 'admins#create'
```
Comment (sagehen03) [inline config/routes.rb:28]: Since we're creating a new record I think this should be a post rather than patch which is for updates.

**Review by sagehen03 — CHANGES_REQUESTED** [review]
Hey Dat,
This is looking good so far--nice job!  You can add tests in `admins_controller_test.rb` and you can see examples of existing controller tests in `clients_controller_test.rb` `donations_controller.test.rb` you can run the tests from the command line using `rails test`

File: app/controllers/admins_controller.rb (hunk 1 above)
Comment (trdtnguyen) [reply app/controllers/admins_controller.rb:11]: Agree. I'll remove it in the next PR.

File: config/routes.rb (hunk 2 above)
Comment (trdtnguyen) [reply config/routes.rb:28]: I'll modify that in the next PR.

**Pull request closed without merging**